src/state/account.ts

```diff
@@ -226,17 +227,32 @@ export const accountSlice = createSlice({
         : [];
 
       // Updates are only considered new after state.subaccount.orders has been set
+      const payloadOrders = action.payload?.orders?.toArray() ?? [];
       const hasNewOrderUpdates =
         state.subaccount?.orders != null &&
-        (action.payload?.orders?.toArray() ?? []).some(
-          (order: SubaccountOrder) => !existingOrderIds.includes(order.id)
-        );
+        payloadOrders.some((order: SubaccountOrder) => !existingOrderIds.includes(order.id));
 
-      if (!state.hasUnseenOrderUpdates) {
-        state.hasUnseenOrderUpdates = hasNewOrderUpdates;
-      }
+      const canceledOrderIdsInPayload = payloadOrders
+        .filter((order) => isOrderStatusCanceled(order.status))
+        .map((order) => order.id);
 
-      state.subaccount = action.payload;
+      return {
+        ...state,
+        subaccount: action.payload,
+        hasUnseenOrderUpdates: hasNewOrderUpdates,
+        localPlaceOrders: canceledOrderIdsInPayload.length
+          ? state.localPlaceOrders.map((order) =>
+              order.submissionStatus < PlaceOrderStatuses.Canceled &&
```

Comment:
hmm something about using comparators with enums always makes me a lil nervous (what if someone reorders) - any thoughts against just using `order.submissionStatus != PlaceOrderStatuses.Canceled`?

Reply:
tru! it originally made more sense with the order of placing/submitted/filled (could explicitly be 0, 1, 2), but now with canceled maybe not -- will update!